### ISSUE TITLE
💅 incorrect fix for `lint/complexity/useOptionalChain` rule.

### ISSUE THREAD

**Issue by BatuhanW** [biomejs/biome#1925]
### Environment information

```bash
> biome rage

CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.19.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.5"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false
```

### Rule name

useOptionalChain

### Playground link

https://biomejs.dev/playground/?importSortingEnabled=false&unsafeParameterDecoratorsEnabled=false&allowComments=false&code=YwBvAG4AcwB0ACAAZgBpAGwAdABlAHIARABlAGEAbABCAGUAZgBvAHIAZQAgAD0AIAAoAGQAZQBhAGwAPwA6ACAARABlAGEAbABBAGcAZwByAGUAZwBhAHQAZQApACAAPQA%2BAAoAIAAgAGQAZQBhAGwAPwAuAGcAcgBvAHUAcABCAHkAIAAmACYAIABkAGUAYQBsAC4AZwByAG8AdQBwAEIAeQA%2FAC4AYwBsAG8AcwBlAEQAYQB0AGUATQBvAG4AdABoACAAJgAmACAAZABlAGEAbAAuAGcAcgBvAHUAcABCAHkAPwAuAGMAbABvAHMAZQBEAGEAdABlAFkAZQBhAHIAOwAKAAoALwAvACAAVABoAGUAIABmAGkAeAAgAGkAcwAgAHMAZQBtAGEAbgB0AGkAYwBhAGwAbAB5ACAAYwBvAHIAcgBlAGMAdAAsACAAaABvAHcAZQB2AGUAcgAsACAAVAB5AHAAZQBzAGMAcgBpAHAAdAAgAGMAbwBtAHAAbABhAGkAbgBzACAAYgBlAGMAYQB1AHMAZQAgAGAAZABlAGEAbABgACAAaQBzACAAcABvAHMAcwBpAGIAbAB5ACAAdQBuAGQAZQBmAGkAbgBlAGQALgAKAC8ALwAgACcAZABlAGEAbAAnACAAaQBzACAAcABvAHMAcwBpAGIAbAB5ACAAJwB1AG4AZABlAGYAaQBuAGUAZAAnAC4ACgBjAG8AbgBzAHQAIABmAGkAbAB0AGUAcgBEAGUAYQBsAEEAcABwAGwAaQBlAGQARgBpAHgAIAA9ACAAKABkAGUAYQBsAD8AOgAgAEQAZQBhAGwAQQBnAGcAcgBlAGcAYQB0AGUAKQAgAD0APgAKACAAIABkAGUAYQBsAC4AZwByAG8AdQBwAEIAeQA%2FAC4AYwBsAG8AcwBlAEQAYQB0AGUATQBvAG4AdABoACAAJgAmACAAZABlAGEAbAAuAGcAcgBvAHUAcABCAHkAPwAuAGMAbABvAHMAZQBEAGEAdABlAFkAZQBhAHIAOwAKAAoALwAvACAASQAgAHQAaABpAG4AawAgAGkAdAAgAHMAaABvAHUAbABkACAAYQBkAGQAIABvAHAAdABpAG8AbgBhAGwAIABjAGgAYQBpAG4AIAB0AG8AIABmAGkAcgBzAHQAIABgAGQAZQBhAGwAYAAgAGMAYQBsAGwALgAKAC8ALwAgAEEAbABzAG8ALAAgAGAAPwBgACAAaQBuACAAcwBlAGMAbwBuAGQAIABlAHgAcAByAGUAcwBzAGkAbwBuACAAYABkAGUAYQBsAC4AZwByAG8AdQBwAEIAeQA%2FAC4AYwBsAG8AcwBlAEQAYQB0AGUAWQBlAGEAcgBgACAAaQBzACAAcgBlAGQAdQBuAGQAYQBuAHQALgAKAC8ALwAgAGkAZgAgAHcAZQAgAGgAYQB2AGUAIAB0AHIAdQB0AGgAeQAgAHYAYQBsAHUAZQAgAGYAbwByACAAYABkAGUAYQBsAD8ALgBnAHIAbwB1AHAAQgB5AD8ALgBjAGwAbwBzAGUARABhAHQAZQBNAG8AbgB0AGgAYAAKAC8ALwAgAFQAaABpAHMAIABtAGUAYQBuAHMAIABgAGQAZQBhAGwALgBnAHIAbwB1AHAAQgB5AGAAIABpAHMAIABuAG8AdAAgAG4AdQBsAGwALgAKAC8ALwAgAFMAbwAgAGkAdAAgAGMAYQBuACAAYgBlACAAYABkAGUAYQBsAC4AZwByAG8AdQBwAEIAeQAuAGMAbABvAHMAZQBEAGEAdABlAFkAZQBhAHIAYAAuAAoAYwBvAG4AcwB0ACAAZgBpAGwAdABlAHIARABlAGEAbABFAHgAcABlAGMAdABlAGQARgBpAHgAIAA9ACAAKABkAGUAYQBsAD8AOgAgAEQAZQBhAGwAQQBnAGcAcgBlAGcAYQB0AGUAKQAgAD0APgAKACAAIABkAGUAYQBsAD8ALgBnAHIAbwB1AHAAQgB5AD8ALgBjAGwAbwBzAGUARABhAHQAZQBNAG8AbgB0AGgAIAAmACYAIABkAGUAYQBsAC4AZwByAG8AdQBwAEIAeQAuAGMAbABvAHMAZQBEAGEAdABlAFkAZQBhAHIAOwA%3D&jsx=false

### Expected result

```typescript
const filterDealExpectedFix = (deal?: DealAggregate) =>
  deal?.groupBy?.closeDateMonth && deal.groupBy.closeDateYear;
```

I think it should add optional chain to first `deal` call.
Also, `?` in second expression `deal.groupBy?.closeDateYear` is redundant.
if we have truthy value for `deal?.groupBy?.closeDateMonth`
This means `deal.groupBy` is not null.
So it can be `deal.groupBy.closeDateYear`.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
I'm not too familiar with this rule yet, but I'm trying to look into it. I can see that your complex test (`foo && foo?.bar && foo?.bar.baz && foo?.bar.baz[buzz] && foo?.bar.baz[buzz]()`) is currently passing, so it appears your initial fix is introducing a regression. I'm also still trying to understand the details of what's going, but my first hunch is that it's going wrong because you're replacing nodes that's not yet expected to do so. I think indeed `replace_node()` should only be done in the action so the code that you modified shouldn't yet do any actual replacement, instead it should transform the suggested replacement expression.

I'll keep looking a bit more to see if this leads me to a solution.

**Comment by Sec-ant** [comment]
Thanks for looking into this! @arendjr 

> so it appears your initial fix is introducing a regression.

Yes, I should've mentioned that this is an existing test case. It's indeed a regression.

> so the code that you modified shouldn't yet do any actual replacement, instead it should transform the suggested replacement expression.

You're right! I think maybe we should reconsider the structure of the signals that are transferred from the `run` part to the `action` part to preserve the lost information from the left part nodes. 

To justify that we should always use the left part of `&&` when applying transformation:

```ts
a?.b && a.b.c // should be fixed as: `a?.b?.c`, use the left part `a?.b`
```

```ts
a.b && a?.b.c // should be fixed as: `a.b?.c`, use the left part `a.b`
// the dot operator in the left part is truthy guaranteed by the `&&` operator
```
